### PR TITLE
feat(ingest): remove source config from DatahubIngestionCheckpoint

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py [hunk 1]
```diff
@@ -220,7 +208,7 @@ def to_checkpoint_aspect(
                 pipelineName=self.pipeline_name,
                 platformInstanceId=self.platform_instance_id,
                 runId=self.run_id,
-                config=self.config.json(),
+                config="",
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py:211]: Can we just remove this??

Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py:211]: (i guess no due to it being a breaking change?)

File: metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py [hunk 2]
```diff
@@ -217,7 +213,6 @@ def create_checkpoint(self) -> Optional[Checkpoint]:
                 pipeline_name=self.pipeline_name,
                 platform_instance_id=self.source.get_platform_instance_id(),
                 run_id=self.run_id,
-                config=cast(ConfigModel, self.config),
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py:220]: What risk do we run by removing this? Does this mean that previously changing a recipe wil create a new checkpoint state completely?

**Review by jjoyce0510 — COMMENTED** [review]
I think the changes look good, my main question is around the implication of changing this.

Does this effectively change the primary key we are using to lookup the previous run state?

File: metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py:220]: Each run of the ingestion recipe will create a new checkpoint state. All this does is not include the raw config as part of that checkpoint

File: metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py:211]: We can't remove it because the field is not optional right now

**Comment by hsheth2** [comment]
This doesn't change the primary key of the checkpoint states - that logic remains the same as before. This has no functional impact, since we weren't making use of these configs anywhere.

**Review by jjoyce0510 — APPROVED** [review]
LGTM

**Pull request merged**